### PR TITLE
protoc enhancements - macos/linux

### PR DESCRIPTION
**Pull request by offerm**
enhance protoc command to make sure it is using the correct plugin and correct syntax
remove unneeded protc package that causes conflics

### CODE REVIEW

**Comment by offerm** [comment]
@sangaman The problem of the 2 missing files caused by package ts-protoc-gen that brings a plugin having the same name of one by grpc_tools_node_protoc_ts.

The change in protoc command line is not really needed (I think) but is there in order to protect against the same problem in the future.

Please help my by checking on Linux.

I didn't change the windows command as I have no idea what would work.

The changes to the generated files are mostly whitespace (can be ignored).

There are some relevant changes.

**Comment by sangaman** [comment]
Ahh great catch, I didn't realize that the `grpc_tools_node_protoc_ts` package also had a `ts-protoc-gen` plugin in it. I'll try this now on both linux and windows and let you know if I have any issues.

**Comment by sangaman** [comment]
So on windows this command:

`"win32": "protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"lib\\proto\"  --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\*.proto proto\\google\\api\\*.proto proto\\google\\protobuf\\*.proto"`

Works perfectly and generates the exact same files as you.

But this command:

`"win32": "node_modules\\grpc-tools\\bin\\protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"lib\\proto\"  --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\*.proto proto\\google\\api\\*.proto proto\\google\\protobuf\\*.proto"`

Gives me an error `proto\*.proto: No such file or directory`. 

On Linux, it works both perfectly both with what you've committed or by taking the `"./node_modules/grpc-tools/bin/` out from the beginning. I believe they actually do the same thing, can we drop that prefix from the proto commands for consistency and brevity? Same for the `swagger` command on mac.

**Comment by offerm** [comment]
Better not.

This way, there is no need for the developer to download and install protoc 3.4. It will be done automatically with `npm i`

try to delete node_modules and rebuild with `npm i`

File: package.json [hunk 1]
```diff
@@ -40,13 +40,13 @@
...
+      "linux":  "./node_modules/grpc-tools/bin/protoc --js_out='import_style=commonjs,binary:lib/proto' --ts_out='lib/proto' --grpc_out='lib/proto' --plugin='protoc-gen-grpc=node_modules/.bin/grpc_tools_node_protoc_plugin' --plugin='protoc-gen-ts=node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts' -I='proto' proto/*.proto proto/google/api/*.proto proto/google/protobuf/*.proto",
+      "darwin": "./node_modules/grpc-tools/bin/protoc --js_out='import_style=commonjs,binary:lib/proto' --ts_out='lib/proto' --grpc_out='lib/proto' --plugin='protoc-gen-grpc=node_modules/.bin/grpc_tools_node_protoc_plugin' --plugin='protoc-gen-ts=node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts' -I='proto' proto/*.proto proto/google/api/*.proto proto/google/protobuf/*.proto",
       "win32": "protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"import_style=commonjs,binary:lib\\proto\"  --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\*.proto proto\\google\\api\\*.proto proto\\google\\protobuf\\*.proto"
```
Comment (sangaman) [inline package.json]: This should be:

`"node_modules\\grpc-tools\\bin\\protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"lib\\proto\" --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\xudrpc.proto proto\\lndrpc.proto proto\\annotations.proto proto\\google\\api\\http.proto proto\\google\\protobuf\\descriptor.proto"`

File: package.json [hunk 2]
```diff
@@ -40,13 +40,13 @@
...
     },
     "swagger": {
       "linux": "protoc --swagger_out='lib/proto' -I='proto' proto/xudrpc.proto",
-      "darwin": "protoc --swagger_out='lib/proto' -I='proto' proto/xudrpc.proto",
+      "darwin": "./node_modules/grpc-tools/bin/protoc --swagger_out='lib/proto' -I='proto' proto/xudrpc.proto",
       "win32": "protoc --swagger_out=\"lib\\proto\" -I=\"proto\" proto\\xudrpc.proto"
```
Comment (sangaman) [inline package.json]: This should be:

`"node_modules\\grpc-tools\\bin\\protoc --swagger_out=\"lib\\proto\" -I=\"proto\" proto\\xudrpc.proto"`

File: package.json [hunk 3]
```diff
@@ -40,13 +40,13 @@
...
+      "darwin": "./node_modules/grpc-tools/bin/protoc --js_out='import_style=commonjs,binary:lib/proto' --ts_out='lib/proto' --grpc_out='lib/proto' --plugin='protoc-gen-grpc=node_modules/.bin/grpc_tools_node_protoc_plugin' --plugin='protoc-gen-ts=node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts' -I='proto' proto/*.proto proto/google/api/*.proto proto/google/protobuf/*.proto",
       "win32": "protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"import_style=commonjs,binary:lib\\proto\"  --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\*.proto proto\\google\\api\\*.proto proto\\google\\protobuf\\*.proto"
     },
     "swagger": {
       "linux": "protoc --swagger_out='lib/proto' -I='proto' proto/xudrpc.proto",
```
Comment (sangaman) [inline package.json]: Have linux use the same command as darwin below, with the qualified `protoc` path

**Pull request merged**